openmc/cmfd.py

```diff
@@ -216,191 +231,181 @@ def map(self, map):
 
         self._map = map
 
-
-    def get_mesh_xml(self):
-
+    def _get_xml_element(self):
         element = ET.Element("mesh")
 
-        if len(self._lower_left) == 2:
-            subelement = ET.SubElement(element, "lower_left")
-            subelement.text = '{0} {1}'.format(self._lower_left[0],
-                                               self._lower_left[1])
-        else:
-            subelement = ET.SubElement(element, "lower_left")
-            subelement.text = '{0} {1} {2}'.format(self._lower_left[0],
-                                                   self._lower_left[1],
-                                                   self._lower_left[2])
-
-        if not self._upper_right is None:
-            if len(self._upper_right) == 2:
-                subelement = ET.SubElement(element, "upper_right")
-                subelement.text = '{0} {1}'.format(self._upper_right[0],
-                                                   self._upper_right[1])
-            else:
-                subelement = ET.SubElement(element, "upper_right")
-                subelement.text = '{0} {1} {2}'.format(self._upper_right[0],
-                                                       self._upper_right[1],
-                                                       self._upper_right[2])
-
-        if len(self._dimension) == 2:
-            subelement = ET.SubElement(element, "dimension")
-            subelement.text = '{0} {1}'.format(self._dimension[0],
-                                               self._dimension[1])
-        else:
-            subelement = ET.SubElement(element, "dimension")
-            subelement.text = '{0} {1} {2}'.format(self._dimension[0],
-                                                   self._dimension[1],
-                                                   self._dimension[2])
-
-        if not self._width is None:
-            if len(self._width) == 2:
-                subelement = ET.SubElement(element, "width")
-                subelement.text = '{0} {1}'.format(self._width[0],
-                                                   self._width[1])
-            else:
-                subelement = ET.SubElement(element, "width")
-                subelement.text = '{0} {1} {2}'.format(self._width[0],
-                                                       self._width[1],
-                                                       self._width[2])
-
-        if not self._energy is None:
+        subelement = ET.SubElement(element, "lower_left")
+        subelement.text = ' '.join(map(str, self._lower_left))
 
-            subelement = ET.SubElement(element, "energy")
+        if self._upper_right is not None:
```

Comment:
don't know if you care, but we should probably use the ``upper_right`` property here as:

```python
if self.upper_right is not None:
```

instead of directly accessing the class member:

```python
if self._upper_right is not None
```

Very minor point, and one that lingers in similar conditionals throughout the API as a carry over from before I switched to properties.